doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+**TODO**:
+ - Plugins
+ - association builder <-> parameters
+
+## Sync
+
+Syncer **SHOULD** synchronise the mapped hardware on parameter change.
+<why>To always keep synchronise the underlined hardware and the PF
+parameters.</why>
+
+Syncer **SHOULD** retrieve parameter value from the hardware if no value has be
+set since the PF start.
+<nb>This is usually implemented on PF start, initialize the parameter values
+with the mapped hardware current state.</nb>
+<why>To allow introspection of the hardware.</why>
+
+A mode with synchronisation on client request **SHOULD** be supported.
+<why>The user may want to group the synchronization of multiple parameters --
+for instance if a syncer contains more than 1 parameter -- in order to avoid
+undesired intermediary states.</why>
+
+Syncers **MAY** report an 'out-of-sync' condition indicating that the hardware
+parameter values are not (or no longer) reflecting the last values set by the
+Parameter Framework.
+<why>This can happen when the underlying hardware subsystem
+crashes/reboots/...</why>
+
+When a syncer reports an out-of-sync condition, the PF **MUST** try to resync
+the hardware values.
+
+# Rule based dynamic abstraction
+
+## Philosophy
+
+The PF offers parameters mapped on hardware. This is a good but weak
+abstraction. There is often a 1/1 relation between a parameter and the hardware
+it maps.
+
+A PF offers a mechanism to abstract the parameters to more high level concept.
+
+The goal is to hide numerous parameters and their dynamic value behind simple
+and human friendly API.
+
+It works by regrouping parameters with similar management and defining
+configurations for each "scenario". These "scenario" are then given a priority
+and a detection predicate. Configuration are applied when their associated
+"scenario" is detected.
+
+"Scenario" are detected through arbitrary criterion provided by the PF host
+(see below).
+
+## Definition
+
+<dl>
+<dt>Configuration</dt>
+<dd>
+Set of values for different parameters. A configuration **MUST NOT** contain 2
+values of the same parameters.
+
+For example, given a PF with 3 integer parameters A,B,C, a configuration can
+contain:
+
+ - 1 value: (A) or (B) or (C); or
+ - 2 values: (A,B) or (A,C) or (B,C); or
+ - 3 values: (A,B,C).
+</dd>
+
+<dt>Rogue Parameter</dt>
+<dd>
+A Parameter that is not contained by any configuration.
+<dd>
+</dl>
+
+## Configuration
+
+A PF **MUST** offer configurations as described in the Definition chapter.
+<nb>rule based parameter engine does not manipulate directly values, it
+applies configuration on the parameters.</nb>
+<why>This is what the reference implementation does.</why>
+
+Each configuration **MUST** be associated with a predicate that condition its
+eligibility. A configuration with a predicate that evaluates to `true` is called
+an "eligible configuration"
+<why>This is what the reference implementation does.</why>
+
+It **SHOULD** be possible to express a predicate to always evaluates to `true`.
```

Comment:
seems to be a trick around implementation. I don't understand the need from a high level standpoint